osidb/models.py

```diff
@@ -723,6 +724,13 @@ def _validate_reported_date(self):
         if self.reported_dt is None:
             raise ValidationError("Flaw has an empty reported_dt")
 
+    def _validate_public_unembargo_date(self):
+        """
+        Check that an unembargo date (public date) exists if the Flaw is public
+        """
+        if not self.is_embargoed and self.unembargo_dt is None:
+            raise ValidationError("Public flaw has an empty unembargo_dt")
```

Comment:
@osoukup is this truly the case? what about 0-days? or do we assume that flaws within Red Hat will **always** start as embargoed and eventually become public?

Reply:
@osoukup while we're at it (maybe this should be a different task) shouldn't we validate that a public flaw's unembargo_dt is always in the past?

Reply:
We always set public date whenever a flaw is not embargoed. For the 0-days we would set public date to the (estimated) time of the first public appearance (possibly predating the reported date).

The second comment is a good point and it makes sense to me to have such validation. That would be also some kind of leak safeguard as OSIDB would not permit an unembargo before the public date (one would have to modify the public date first so that would make her/him think twice). I know this is above the original task but could you @costaconrado add this validation too? It should be fairly similar.

Reply:
Oh I see, then I think I should change the factory itself to something like 
```
    unembargo_dt = factory.Maybe(
        "embargoed",
        yes_declaration=factory.Faker("future_datetime", tzinfo=UTC),
        no_declaration=factory.Faker("past_datetime", tzinfo=UTC),
    )
```

I will create this new validation and change the factory to reflect it